### ISSUE TITLE
Pub Sub Sample Feature - one long line of concatenated json objects is suboptimal

### ISSUE THREAD

**Issue by robert-hanuschke** [awslabs/aws-iot-device-client#403]
**Describe the bug**

I successfully set up the [Pub Sub Sample](https://github.com/awslabs/aws-iot-device-client/blob/main/source/samples/pubsub/README.md) and my client writes the JSON payload I send to the file specified.
However, it appears there as one singular line of concatenated JSON objects.

Like: 
`{"key": "value"}{"key": "value"}{"key": "value"}`

This is suboptimal for working with the payload. One has to split the line by "}{" and work with the result. And even then, either you read the whole line (which can be a very hefty size after lots of communication) or you decide on a length to cut from the end - which bears the risk to not have a full object at all and requires guesswork on the user's end.

**To Reproduce**

1. Set up Pub Sub Sample
2. Send some JSON payloads to the subscribe topic
3. Check the subscribe-file to see singular line being concatenated to

**Expected behavior**

Each received payload is written to a new line in the file. Like this, you just read n last lines and each is a valid JSON object on its own.

**Actual behavior**

One single line of concatenated JSON objects, making working with the data cumbersome.

**Logs**

If applicable, add full logs of errors and outputs to help explain your problem. Preferably, you can also [increase the verbosity of the logs](https://github.com/awslabs/aws-iot-device-client#logging).

**Environment (please complete the following information):**
 - OS: Raspbian
 - Version: 11 (bullseye)
 - Architecture: armv7l
 - Device Client commit: don't have the commit, but built from the latest release. aws-iot-device-client --version is v1.8.18-6633b39

**Additional context**

**Comment by HarshGandhi-AWS** [comment]
Hello @robert-hanuschke , thank you for reaching out to us. I am looking into this issue now. Looks like [over here](https://github.com/awslabs/aws-iot-device-client/blob/main/source/samples/pubsub/PubSubFeature.cpp#L273) we should add newline char after writing payload data in local file. 

I am not sure if adding newline character is what every user wants. For payload larger than MQTT message size or user's who are sending text (not in JSON format) might not wanna have newline char added at the end of each payload message in the file. 

I will get back to you after consulting with my team to understand if this change is required or not. Till then to unblock yourself, I would suggest you to manually make changes in the code to support your business case. 

Thank you, 
Harsh Gandhi

**Comment by HarshGandhi-AWS** [comment]
Since it is an expected behavior this is not a bug. Adding `enhancement` label instead.

**Comment by HarshGandhi-AWS** [comment]
Hello @robert-hanuschke, I spoke with my team and we think while publishing messages from cloud side to store in `subscribe-file` on the device side you can add the newline character in message payload. Device Client will read and store message payload with the newline char in the file.
This way all new mqtt messages will be stored on newline as you want. 

Regards,
Harsh Gandhi

**Comment by HarshGandhi-AWS** [comment]
I am closing this issue now. Please feel free  to reopen this issue or create new one if you have any other questions.
